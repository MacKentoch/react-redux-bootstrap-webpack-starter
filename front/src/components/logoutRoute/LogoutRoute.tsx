import React, { useEffect } from 'react';
import { Route, Redirect } from 'react-router-dom';
import { RouteComponentProps } from 'react-router';
import { ReduxConnectedProps, OwnProps } from './index';

// #region types
type Props = RouteComponentProps & ReduxConnectedProps & OwnProps;
// #endregion

function LogoutRoute(props: Props) {
  const { disconnectUser } = props;
  useEffect(() => disconnectUser());

  return (
    <Route {...props}>
      <Redirect to={{ pathname: '/login' }} />
    </Route>
  );
}

LogoutRoute.displayName = 'LogoutRoute';

export default LogoutRoute;
