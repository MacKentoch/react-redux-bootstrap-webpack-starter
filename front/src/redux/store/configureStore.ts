import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { createBrowserHistory as createHistory } from 'history';
import { persistReducer, persistStore } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import reducer from '../modules/reducers';

// #region constants
const isProd = process.env.NODE_ENV === 'production';
export const history = createHistory();
// #endregion

// #region createStore : enhancer

// #region logger middleware (dev only)
const loggerMiddleware = createLogger({
  level: 'info',
  collapsed: true,
});
// #endregion

const enhancer = !isProd
  ? composeWithDevTools(
      applyMiddleware(
        thunkMiddleware,
        loggerMiddleware, // logger at the end
      ),
    )
  : composeWithDevTools(applyMiddleware(thunkMiddleware));
// #endregion

// #region persisted reducer
const persistConfig = {
  key: 'root',
  storage,
  blacklist: ['router'],
  // whitelist: ['userAuth'],
};

const persistedReducer = persistReducer(persistConfig, reducer);
// #endregion

export default function configureStore(initialState = {}) {
  const store = createStore(persistedReducer, initialState, enhancer);
  const persistor = persistStore(store);

  if ((module as any).hot) {
    (module as any).hot?.accept('../modules/reducers', () => {
      // @ts-ignore
      const nextAppReducer = require('../modules/reducers').default;
      store.replaceReducer(persistReducer(persistConfig, nextAppReducer));
    });
  }

  return { store, persistor };
}
