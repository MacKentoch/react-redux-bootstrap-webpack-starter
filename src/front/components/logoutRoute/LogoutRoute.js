// @flow

// #region imports
import React, { PureComponent } from 'react';
import { Route, Redirect, withRouter } from 'react-router-dom';
import {
  type Match,
  type Location,
  type RouterHistory,
} from 'react-router-dom';
import auth from '../../services/auth';
// #endregion

// #region flow types
type Props = {
  // react-router 4:
  match: Match,
  location: Location,
  history: RouterHistory,

  ...any,
};
type State = any;
// #endregion

class LogoutRoute extends PureComponent<Props, State> {
  // #region lifecycle
  componentDidMount() {
    auth.clearAllAppStorage();
  }

  render() {
    return (
      <Route {...this.props}>
        <Redirect to={{ pathname: '/login' }} />
      </Route>
    );
  }
  // #endregion
}

export default withRouter(LogoutRoute);
