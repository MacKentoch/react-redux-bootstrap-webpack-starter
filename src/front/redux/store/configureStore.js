// @flow

/* eslint no-process-env:0 */
if (process.env.NODE_ENV === 'production') {
  module.exports = require('./configureStore.prod');
} else {
  module.exports = require('./configureStore.dev');
}
