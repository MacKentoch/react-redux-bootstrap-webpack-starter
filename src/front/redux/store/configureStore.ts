/* eslint no-process-env:0 */
let configureStore = null;

if (process.env.NODE_ENV === 'production') {
  configureStore = require('./configureStore.prod').default;
  // const {history} = require('./configureStore.prod');
  // export history;
} else {
  configureStore = require('./configureStore.dev').default;
  // const {history} = require('./configureStore.dev');
  // export history;
}
export default configureStore;
