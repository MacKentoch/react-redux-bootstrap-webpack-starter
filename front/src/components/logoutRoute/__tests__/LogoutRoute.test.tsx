import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter, Switch, Route } from 'react-router';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';
import configureStore from 'redux-mock-store';
import LogoutRoute from '../index';

const middlewares: Array<any> = [];
const mockStore = configureStore(middlewares);

describe('LogoutRoute component', () => {
  let rootElement: any = null;

  beforeEach(() => {
    rootElement = document.createElement('div');
    document.body.appendChild(rootElement);
  });

  afterEach(() => {
    rootElement && document.body.removeChild(rootElement);
    rootElement = null;
  });

  it('renders as expected', async () => {
    const initialState = {
      userAuth: {
        isFetching: false,
        isLogging: false,
        id: '',
        login: '',
        firstname: '',
        lastname: '',
        token: '',
        isAuthenticated: false,
      },
    };
    const store = mockStore(initialState);
    const { container } = await render(
      <Provider store={store}>
        <ThemeProvider theme={{}}>
          <MemoryRouter>
            <Switch>
              <Route path="*">
                <span>anywhere</span>
              </Route>
              <Route path="/login">
                <span> login page</span>
              </Route>
              <LogoutRoute />
            </Switch>
          </MemoryRouter>
        </ThemeProvider>
      </Provider>,
      rootElement,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
