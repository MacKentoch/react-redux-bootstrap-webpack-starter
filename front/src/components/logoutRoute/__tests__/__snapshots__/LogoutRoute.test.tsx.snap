// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoutRoute component renders as expected 1`] = `
<span>
  anywhere
</span>
`;
